libcaf_core/test/detail/rfc3629.cpp

```diff
@@ -122,43 +123,87 @@ constexpr std::string_view ascii_2 = R"__(
+  check(valid_utf8(valid_four_byte_1));
+  check(valid_utf8(valid_four_byte_2));
+}
+
+TEST("invalid UTF-8 input") {
+  check(!valid_utf8(invalid_two_byte_1));
+  check(!valid_utf8(invalid_two_byte_2));
+  check(!valid_utf8(invalid_two_byte_3));
+  check(!valid_utf8(invalid_two_byte_4));
+  check(!valid_utf8(invalid_three_byte_1));
+  check(!valid_utf8(invalid_three_byte_2));
+  check(!valid_utf8(invalid_three_byte_3));
+  check(!valid_utf8(invalid_three_byte_4));
+  check(!valid_utf8(invalid_three_byte_5));
+  check(!valid_utf8(invalid_three_byte_6));
+  check(!valid_utf8(invalid_three_byte_7));
+  check(!valid_utf8(invalid_three_byte_8));
+  check(!valid_utf8(invalid_four_byte_1));
+  check(!valid_utf8(invalid_four_byte_2));
+  check(!valid_utf8(invalid_four_byte_3));
+  check(!valid_utf8(invalid_four_byte_4));
+  check(!valid_utf8(invalid_four_byte_5));
+  check(!valid_utf8(invalid_four_byte_6));
+  check(!valid_utf8(invalid_four_byte_7));
+  check(!valid_utf8(invalid_four_byte_8));
+  check(!valid_utf8(invalid_four_byte_9));
+  check(!valid_utf8(invalid_four_byte_10));
+}
+
+TEST("invalid UTF-8 input missing continuation bytes") {
+  check_eq(rfc3629::validate(invalid_two_byte_1), res_t(0, true));
+  check_eq(rfc3629::validate(invalid_three_byte_1), res_t{0ul, true});
+  check_eq(rfc3629::validate(invalid_three_byte_2), res_t{0ul, true});
+  check_eq(rfc3629::validate(invalid_four_byte_1), res_t{0ul, true});
+  check_eq(rfc3629::validate(invalid_four_byte_2), res_t{0ul, true});
+  check_eq(rfc3629::validate(invalid_four_byte_3), res_t{0ul, true});
+}
+
+TEST("invalid UTF-8 input malformed data") {
+  check_eq(rfc3629::validate(invalid_two_byte_2), res_t{0ul, false});
+  check_eq(rfc3629::validate(invalid_two_byte_3), res_t{0ul, false});
+  check_eq(rfc3629::validate(invalid_two_byte_4), res_t{0ul, false});
+  check_eq(rfc3629::validate(invalid_three_byte_3), res_t{0ul, false});
+  check_eq(rfc3629::validate(invalid_three_byte_4), res_t{0ul, false});
+  check_eq(rfc3629::validate(invalid_three_byte_5), res_t{0ul, false});
+  check_eq(rfc3629::validate(invalid_three_byte_6), res_t{0ul, false});
+  check_eq(rfc3629::validate(invalid_three_byte_7), res_t{0ul, false});
+  check_eq(rfc3629::validate(invalid_three_byte_8), res_t{0ul, false});
+  check_eq(rfc3629::validate(invalid_four_byte_4), res_t{0ul, false});
+  check_eq(rfc3629::validate(invalid_four_byte_5), res_t{0ul, false});
+  check_eq(rfc3629::validate(invalid_four_byte_6), res_t{0ul, false});
+  check_eq(rfc3629::validate(invalid_four_byte_7), res_t{0ul, false});
+  check_eq(rfc3629::validate(invalid_four_byte_8), res_t{0ul, false});
+  check_eq(rfc3629::validate(invalid_four_byte_9), res_t{0ul, false});
+  check_eq(rfc3629::validate(invalid_four_byte_10), res_t{0ul, false});
+}
 
-TEST_CASE("ASCII input") {
-  CHECK(valid_utf8(ascii_1));
-  CHECK(valid_utf8(ascii_2));
+TEST("invalid UTF-8 input fails on the invalid byte") {
+  const_byte_span input{invalid_four_byte_9};
```

Comment:
Coding convention: prefer `auto`:

```suggestion
  auto input = make_span(invalid_four_byte_9);
```